### ISSUE TITLE
Unable to create aws_waf_ipset from a list of IPs

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#344]
_This issue was originally opened by @jmulloy-simplisafe as hashicorp/terraform#9814. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

I would like to use terraform to load an AWS WAF whitelist and load the same IP whitelist into consul so our API apps can pick it up to enforce access by IP. I can't find any way to load a list of IPs into an aws_waf_ipset resource. This problem seems to have been solved for security groups by adding a separate aws_security_group_rule resource that can be iterated via count. Perhaps this would be the easiest solution for this problem.

I've also been trying to do the reverse, getting the IPs out of a manually configured aws_waf_ipset resource and turning it into a list that I could load into consul, but I can't get that to work either.

Also the documentation for this resource doesn't make it clear that repeating the stanza is how you add multiple IPs to the resource.

### Terraform Version
0.7.8

### Affected Resource(s)
- aws_waf_ipset

### Terraform Configuration Files
```hcl
resource "aws_waf_ipset" "foobar_whitelist" {
  name = "tf-whitelist-test"
  ip_set_descriptors {
    type = "IPV4"
    value = "10.10.10.10/32"
  }
  ip_set_descriptors {
    type = "IPV4"
    value = "20.20.20.20/32"
  }
}
```

**Comment by AlexRudd** [comment]
This is a problem for me also

**Comment by shatil** [comment]
This affects more than a few constraints or conditions in WAF, including `byte_match_tuples` in `aws_wafregional_byte_match_set`, where I can't merely use `count` to add additional `byte_match_tuples`. The current design is frustrating to use.

**Comment by bflad** [comment]
Hi folks 👋 This issue is resolved in Terraform 0.12, which supports new functionality in the configuration language aimed at solving this issue. The new [`dynamic` block syntax](https://www.terraform.io/docs/configuration/expressions.html#dynamic-blocks) can be used to dynamically generate configuration blocks and their arguments.

Given this configuration:

```hcl
terraform {
  required_providers {
    aws = "2.20.0"
  }
  required_version = "0.12.5"
}

provider "aws" {
  region = "us-east-1"
}

resource "aws_waf_ipset" "test" {
  name = "test"

  dynamic "ip_set_descriptors" {
    # The for_each argument is a hardcoded list in this illustrative example,
    # however it can be sourced from a variable or local value as well as
    # support multiple argument values as a map.
    for_each = ["1.1.1.1/32", "2.2.2.2/32", "3.3.3.3/32"]

    content {
      type  = "IPV4"
      value = ip_set_descriptors.value
    }
  }
}

output "test" {
  value = aws_waf_ipset.test.ip_set_descriptors[*].value
}
```

Produces the following apply output:

```
$ terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_waf_ipset.test will be created
  + resource "aws_waf_ipset" "test" {
      + arn  = (known after apply)
      + id   = (known after apply)
      + name = "test"

      + ip_set_descriptors {
          + type  = "IPV4"
          + value = "1.1.1.1/32"
        }
      + ip_set_descriptors {
          + type  = "IPV4"
          + value = "2.2.2.2/32"
        }
      + ip_set_descriptors {
          + type  = "IPV4"
          + value = "3.3.3.3/32"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_waf_ipset.test: Creating...
aws_waf_ipset.test: Creation complete after 1s [id=68eb652e-c310-41c7-9f2e-9e8509c91f46]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

test = [
  "1.1.1.1/32",
  "2.2.2.2/32",
  "3.3.3.3/32",
]
```

Enjoy! 🚀
